.github/workflows/rust.yml

```diff
@@ -38,20 +41,37 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
+        with:
+          components: clippy
+      - uses: taiki-e/install-action@v2
+        with:
+          tool: cargo-nextest
       - uses: Swatinem/rust-cache@v2
-      - run: cd ${{ matrix.package }} && cargo test --all-targets
+      - run: cd ${{ matrix.package }} && cargo nextest run
       - run: cd ${{ matrix.package }} && cargo clippy --all-targets -- -D warnings
 
-  Lint:
+  Rustfmt:
     runs-on: ubuntu-latest
     steps:
+      # No need to test `rinja_derive_standalone`. It has same the `src` folder as `rinja_derive`.
+      # No need to run the checks in parallel. They are fast enough.
+      # No need for caching. No code is compiled at all.
       - uses: actions/checkout@v4
-      - uses: dtolnay/rust-toolchain@stable
+      - uses: dtolnay/rust-toolchain@nightly
         with:
-          components: rustfmt, clippy
-      - uses: Swatinem/rust-cache@v2
-      - run: cargo fmt --all -- --check
-      - run: cargo clippy --all-targets -- -D warnings
+          components: rustfmt
+      - run: |
+          set -eu
+          for PKG in \
+            rinja rinja_actix rinja_axum rinja_derive rinja_derive_standalone \
+            rinja_parser rinja_rocket rinja_warp testing examples/actix-web-app
+          do
+            cd "$PKG"
+            echo "Testing: $PKG"
+            cargo fmt -- --check
```

Comment:
Just one question: there is no more `clippy` check here. However I see there is one above. Is `--all-targets` run on all crates even if they are in different workspaces?

Reply:
Does `--all-targets` even do anything if you don't specify a target list somewhere? Isn't all=native in by default?

Clippy checking on +nightly is always stressful to me, because often tests are a bit over-eager in their early stage.

The clippy further up ensures that all packages are tested on +stable. If you run clippy in a workspace root, I think normally only default-members are tested, but I'm not sure.

Reply:
Hum... I don't mind running non-nightly clippy. Just want to be sure it'll check all crates, even those which are not in the default workspace.

Reply:
The list is exhaustive. Everything is checked. We just have to make sure, that the list stays exhaustive. I don't see use adding new crates every other day, so this requirement should be fine. :)

Reply:
Sorry, that's not what I meant. Let me try again. 😆 

So now, the default workspace doesn't include all crates. So here, I was wondering if `cargo clippy --all-targets` was checking all crates, even if not in the default workspace. If not, could we make the same loop (and store the list of crate in a variable?) so we can run `clippy` (stable version) on all crates?

Reply:
I confused myself what "targets" are in this case. In other cases "target" is short for "target platform", but in here "target" is the same as in Make targets: 

> Build all targets. This is equivalent to specifying --lib --bins --tests --benches --examples.

No, clippy won't traverse into folders to find additional crates. If clippy is executed in a workspace root, the workspace is checked (I don't know if all members or only the default-members). If the current folder is not a workspace root, then the current package is tested.

In the **current** configuration **not all** packages are linted. In the **new** configuration **all** packages are linted. The new config is strictly better than the old one. :)

Reply:
So confusing. XD Thanks for the clarification! Then all good for me. :)